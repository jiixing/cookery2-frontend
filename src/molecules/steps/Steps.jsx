// @flow

import React from 'react'

import { Col, Card, CardHeader, ListGroup, ListGroupItem } from 'reactstrap'
import Step from 'atoms/step'

import './styles.css'

import type { Step as StepType } from 'types'

const renderStep = (step: StepType) => (
  <ListGroupItem
    key={step.isRecipe ? step.name : step.description}
    className="step"
  >
    <Step {...step} />
  </ListGroupItem>
)

const Steps = ({ items }: { items: Array<StepType> }) => (
  <Col xs="12" lg="6" className="mb-3">
    <Card>
      <CardHeader>
        <h6>Steps</h6>
      </CardHeader>
      <ListGroup flush>{items.map(step => renderStep(step))}</ListGroup>
    </Card>
  </Col>
)

Steps.defaultProps = { items: [] }

export default Steps
