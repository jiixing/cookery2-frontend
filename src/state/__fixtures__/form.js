const form = () => ({
  recipe: {
    name: '',
    difficulty: 'easy',
    servings: 1,
    duration: 0,
  },
  forms: {
    $form: {
      initialValue: {
        recipe: {
          name: '',
          difficulty: 'easy',
          servings: 1,
          duration: 0,
        },
      },
      focus: false,
      pending: false,
      pristine: true,
      submitted: false,
      submitFailed: false,
      retouched: false,
      touched: false,
      valid: true,
      validating: false,
      validated: false,
      validity: {},
      errors: {},
      intents: [],
      model: 'form',
      value: {
        recipe: {
          name: '',
          difficulty: 'easy',
          servings: 1,
          duration: 0,
        },
      },
    },
    recipe: {
      $form: {
        initialValue: {
          name: '',
          difficulty: 'easy',
          servings: 1,
          duration: 0,
        },
        focus: false,
        pending: false,
        pristine: true,
        submitted: false,
        submitFailed: false,
        retouched: false,
        touched: false,
        valid: true,
        validating: false,
        validated: false,
        validity: {},
        errors: {},
        intents: [],
        model: 'form.recipe',
        value: {
          name: '',
          difficulty: 'easy',
          servings: 1,
          duration: 0,
        },
      },
      name: {
        initialValue: '',
        focus: false,
        pending: false,
        pristine: true,
        submitted: false,
        submitFailed: false,
        retouched: false,
        touched: false,
        valid: true,
        validating: false,
        validated: false,
        validity: {},
        errors: {},
        intents: [],
        model: 'form.recipe.name',
        value: '',
      },
      difficulty: {
        initialValue: 'easy',
        focus: false,
        pending: false,
        pristine: true,
        submitted: false,
        submitFailed: false,
        retouched: false,
        touched: false,
        valid: true,
        validating: false,
        validated: false,
        validity: {},
        errors: {},
        intents: [],
        model: 'form.recipe.difficulty',
        value: 'easy',
      },
      servings: {
        initialValue: 1,
        focus: false,
        pending: false,
        pristine: true,
        submitted: false,
        submitFailed: false,
        retouched: false,
        touched: false,
        valid: true,
        validating: false,
        validated: false,
        validity: {},
        errors: {},
        intents: [],
        model: 'form.recipe.servings',
        value: 1,
      },
      duration: {
        initialValue: 0,
        focus: false,
        pending: false,
        pristine: true,
        submitted: false,
        submitFailed: false,
        retouched: false,
        touched: false,
        valid: true,
        validating: false,
        validated: false,
        validity: {},
        errors: {},
        intents: [],
        model: 'form.recipe.duration',
        value: 0,
      },
    },
  },
})

export default form
