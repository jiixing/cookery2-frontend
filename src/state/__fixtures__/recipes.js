const recipes = () => ({
  entities: {
    recipes: {
      1: {
        id: 1,
        name: 'test name',
        servings: 2,
        difficulty: 'easy',
        duration: 0,
      },
      2: {
        id: 2,
        name: 'Bell’s Expedition',
        servings: 5,
        difficulty: 'easy',
        duration: 0,
      },
      3: {
        id: 3,
        name: 'Founders Kentucky Breakfast ',
        servings: 2,
        difficulty: 'medium',
        duration: 0,
      },
      4: {
        id: 4,
        name: 'Chimay Grande Réserve',
        servings: 3,
        difficulty: 'medium',
        duration: 0,
      },
      5: {
        id: 5,
        name: 'Alpha King Pale Ale',
        servings: 1,
        difficulty: 'difficult',
        duration: 0,
      },
      6: {
        id: 6,
        name: 'La Fin Du Monde',
        servings: 6,
        difficulty: 'easy',
        duration: 0,
      },
      7: {
        id: 7,
        name: 'Trappistes Rochefort 10',
        servings: 1,
        difficulty: 'difficult',
        duration: 0,
      },
      10: {
        id: 10,
        name: 'Yeti Imperial Stout',
        servings: 2,
        difficulty: 'medium',
        duration: 0,
      },
      11: {
        id: 11,
        name: 'Samuel Smith’s Oatmeal Stout',
        servings: 6,
        difficulty: 'medium',
        duration: 93,
      },
      15: {
        id: 15,
        name: 'my new thing',
        servings: 3,
        difficulty: 'easy',
        duration: 0,
      },
      16: {
        id: 16,
        name: 'name name name',
        servings: 3,
        difficulty: 'medium',
        duration: 0,
      },
    },
  },
  result: [1, 2, 3, 4, 5, 6, 7, 10, 11, 15, 16],
})

export default recipes
