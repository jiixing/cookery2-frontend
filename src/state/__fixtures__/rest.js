export const recipes = [
  {
    id: 1,
    name: 'test name',
    servings: 2,
    difficulty: 'easy',
    duration: 0
  },
  {
    id: 2,
    name: 'Bell’s Expedition',
    servings: 5,
    difficulty: 'easy',
    duration: 0
  },
  {
    id: 3,
    name: 'Founders Kentucky Breakfast ',
    servings: 2,
    difficulty: 'medium',
    duration: 0
  },
  {
    id: 4,
    name: 'Chimay Grande Réserve',
    servings: 3,
    difficulty: 'medium',
    duration: 0
  },
  {
    id: 5,
    name: 'Alpha King Pale Ale',
    servings: 1,
    difficulty: 'difficult',
    duration: 0
  },
  {
    id: 6,
    name: 'La Fin Du Monde',
    servings: 6,
    difficulty: 'easy',
    duration: 0
  },
  {
    id: 7,
    name: 'Trappistes Rochefort 10',
    servings: 1,
    difficulty: 'difficult',
    duration: 0
  },
  {
    id: 10,
    name: 'Yeti Imperial Stout',
    servings: 2,
    difficulty: 'medium',
    duration: 0
  },
  {
    id: 11,
    name: 'Samuel Smith’s Oatmeal Stout',
    servings: 6,
    difficulty: 'medium',
    duration: 93
  },
  {
    id: 15,
    name: 'my new thing',
    servings: 3,
    difficulty: 'easy',
    duration: 0
  },
  {
    id: 16,
    name: 'name name name',
    servings: 3,
    difficulty: 'medium',
    duration: 0
  }
]

export const recipe = {
  quantities: [
    { name: 'sugar', unit: 'gr', amount: 50 },
    { name: 'water', unit: 'ml', amount: 1000 },
    { name: 'salt', unit: 'pinch', amount: 2 }
  ],
  steps: [
    { isRecipe: false, description: 'Mix all the ingredients' },
    { isRecipe: true, id: 69, name: 'Other' }
  ],
  ...recipes[0]
}
