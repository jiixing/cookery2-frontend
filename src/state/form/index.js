import * as constants from './constants'

export { constants }
export { default as reducer } from './reducers'

export { modelName } from './utils'
