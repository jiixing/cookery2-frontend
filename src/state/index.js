import { combineReducers } from 'redux'

import {
  constants as recipesConstants,
  reducer as recipesReducer,
} from './recipes'
import {
  constants as recipeConstants,
  reducer as recipeReducer,
} from './recipe'

const rootReducer = combineReducers({
  [recipesConstants.namespace]: recipesReducer,
  [recipeConstants.namespace]: recipeReducer,
})

export default rootReducer
