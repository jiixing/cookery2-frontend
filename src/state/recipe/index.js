import * as constants from './constants'
import * as actions from './actions'

export { actions, constants }
export { default as recipeSelector } from './selectors'
export { default as reducer } from './reducers'
