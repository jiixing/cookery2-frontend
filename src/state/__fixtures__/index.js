import recipes from './recipes'
import * as rest from './rest'

const state = { recipes: recipes() }

export { recipes, rest }
export default () => state
