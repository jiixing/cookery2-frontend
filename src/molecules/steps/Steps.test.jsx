import React from 'react'
import { fullRender } from 'test'

import Steps from './Steps'

import { recipe } from 'state/__fixtures__'

describe('components', () => {
  describe('Steps', () => {
    it('renders correctly', () => {
      const { component } = fullRender(<Steps items={recipe().steps} />)
      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
