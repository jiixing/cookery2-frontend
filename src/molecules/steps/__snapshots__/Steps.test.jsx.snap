// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Steps renders correctly 1`] = `
<div
  className="mb-3 col-12 col-lg-6"
>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      <h4>
        Steps
      </h4>
    </div>
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="step list-group-item"
      >
        <span>
          <div>
            <span>
              Mix all the ingredients
            </span>
          </div>
        </span>
      </li>
      <li
        className="step list-group-item"
      >
        <span>
          <a
            href="/recipes/69"
            onClick={[Function]}
          >
            Other
          </a>
        </span>
      </li>
    </ul>
  </div>
</div>
`;
